fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.postgresql;
+
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.ConnectionDetails;
+import com.ibm.fhir.database.utils.api.ConnectionException;
+import com.ibm.fhir.database.utils.api.DataAccessException;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.api.LockException;
+import com.ibm.fhir.database.utils.api.UndefinedNameException;
+import com.ibm.fhir.database.utils.api.UniqueConstraintViolationException;
+
+/**
+ * translates database access to PostgreSql supported access.
+ */
+public class PostgreSqlTranslator implements IDatabaseTranslator {
+    private static final Logger logger = Logger.getLogger(PostgreSqlTranslator.class.getName());
+
+    @Override
+    public String addForUpdate(String sql) {
+        return sql + " FOR UPDATE";
+    }
+
+    @Override
+    public boolean isDerby() {
+        return false;
+    }
+
+    @Override
+    public String globalTempTableName(String tableName) {
+        return "SYSTEM." + tableName;
+    }
+
+    @Override
+    public String createGlobalTempTable(String ddl) {
+        return "DECLARE " + ddl;
+    }
+
+    @Override
+    public boolean isDuplicate(SQLException x) {
+        // Class Code 23: Constraint Violation
```

Comment:
```suggestion
        // Class Code 23: Constraint Violation
        // Refer to https://www.postgresql.org/docs/12/errcodes-appendix.html for more detail
```

I always want to know more, the link will help me (since I'm new )

Reply:
It could be referenced at the top of the class.